pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook_test.go

```diff
@@ -518,6 +518,26 @@ func TestHandleMemberCluster(t *testing.T) {
 		resourceValidator fleetResourceValidator
 		wantResponse      admission.Response
 	}{
+		"allow create upstream fleet MC by any user": {
```

Comment:
Hi Arvind! We probably also need a UT for verifying the deletion behavior and another for verifying the `FleetAnnotationPrefix` presence branch.

Reply:
I understand that we might have already covered some of these in E2Es (upstream/RP)?

Reply:
The cases are defined as a E2E https://github.com/Azure/fleet/blob/main/test/e2e/fleet_guard_rail_test.go#L49-L488

Added this case as I noticed we were not checking the allowed Message anywhere